src/main/java/seedu/address/logic/parser/MarkCommandParser.java

```diff
@@ -25,17 +30,23 @@ public class MarkCommandParser implements Parser<MarkCommand> {
     @Override
     public MarkCommand parse(String args) throws ParseException {
         ArgumentMultimap argMultimap =
-                ArgumentTokenizer.tokenize(args, PREFIX_WEEK, PREFIX_INDICES);
-        if (!arePrefixesPresent(argMultimap, PREFIX_WEEK, PREFIX_INDICES)
+                ArgumentTokenizer.tokenize(args, PREFIX_WEEK, PREFIX_PRESENT, PREFIX_ABSENT);
+
+        if (!arePrefixesPresent(argMultimap, PREFIX_WEEK)
+            || (!arePrefixesPresent(argMultimap, PREFIX_PRESENT) && !arePrefixesPresent(argMultimap, PREFIX_ABSENT))
+            || (arePrefixesPresent(argMultimap, PREFIX_PRESENT) && argMultimap.getValue(PREFIX_PRESENT).get().isEmpty())
+            || (arePrefixesPresent(argMultimap, PREFIX_ABSENT) && argMultimap.getValue(PREFIX_ABSENT).get().isEmpty())
                 || !argMultimap.getPreamble().isEmpty()) {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, MarkCommand.MESSAGE_USAGE));
         }
-
-        argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_WEEK, PREFIX_INDICES);
+        argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_WEEK, PREFIX_PRESENT, PREFIX_ABSENT);
         Week week = ParserUtil.parseWeek(argMultimap.getValue(PREFIX_WEEK).get());
-        List<Index> indices = ParserUtil.parseIndices(argMultimap.getValue(PREFIX_INDICES).get());
-
-        return new MarkCommand(week, indices);
+        List<Index> presentIndices = ParserUtil.parseIndices(argMultimap.getValue(PREFIX_PRESENT).orElse(""));
+        List<Index> absentIndices = ParserUtil.parseIndices(argMultimap.getValue(PREFIX_ABSENT).orElse(""));
```

Comment:
Can we make it so that parseIndices doesn't accept empty list of indices and doesnt return an empty list, but rather make it so that the prefix itself can be non existent (like how we dont need telegram prefix in add contact)? So if the person writes pre/ prefix, then the indices have to follow. Cuz i think returning an empty list was something we disallowed in one of the tests, forgot which

On second thoughts, (as i was typing out the above para), i realised empty list is fine, since our MarkCommand will just not iterate over anything. What do you think?

Reply:
Yup so actually, the reason I do this is because we can't have one of the lists in MarkCommand uninitialised, so I need to pass the constructor something. Plus I tested it out and when the list is empty what happens is it just displays an empty list (because toString etc works perfectly on it). 

What do you think, is it fine to leave it this way? Or if you have a better way I can change it too

Reply:
OK thats perfect, soudns good approving now